### PR TITLE
svg logos

### PR DESCRIPTION
**Pull request by gmarinov**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-7297

this PR:
- enables support for SVG logos
- refactors logo config into yaml file
- exports a manifest/config files for brandingtool to read and access static assets bucket (for SVG files)
- adds a test to check for redundant or missing logo assets
- adds SVG logos for local radio stations

notes:
- see https://github.com/bbc/brandingtool-cloud/pull/182
- framework.yaml references _int_ S3 path for testing purposes. needs changing before merge.
- onair panel needs changing in separate repo.

1 desktop_header

1 desktop_footer

1 mobile_header

1 mobile_footer

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -120,21 +122,56 @@ gulp.task('images', gulp.series('images:clean', function() {
...
+    // @todo clean rev-logos for env
     del([manifestPath.join('/')]),
-    del([`${staticPathDist}/rev-manifest.json`])
+    del([`${staticPathDist}/rev-manifest.json`]),
+    del([`${staticPathDist}/rev-logos.json`])
 ]));
 
+
+gulp.task('rev:logoconfig', function(done) {
+    const srcParameters = './config/parameters.yaml'
```
Comment (Tasiobg) [inline gulpfile.js:133]: Is srcParameters used anywhere?

Comment (gmarinov) [reply gulpfile.js:133]: ermm, no, nice catch.

File: src/Twig/NetworkLogoExtension.php [hunk 2]
```diff
@@ -69,10 +44,27 @@ public function getNetworkLogo(string $nid, string $size): string
...
+        $logo = ($this->isAllowedLogo($nid)) ? $nid : 'bbc';
+
+        if ($this->isSvgLogo($nid)) {
+            array_push($path, 'svg/', $logo, '.svg');
+        } else {
+            array_push($path, $size, '/', $logo, '.png');
         }
-        return $this->packages->getUrl('images/logos/' . $size . '/' . $logo . '.png');
+
+        return $this->packages->getUrl(join('', $path)) . '?mt' . microtime(true);
```
Comment (gmarinov) [inline src/Twig/NetworkLogoExtension.php:56]: @todo remove cache-busting

File: src/Twig/NetworkLogoExtension.php [hunk 3]
```diff
@@ -69,10 +44,33 @@ public function getNetworkLogo(string $nid, string $size): string
...
+
+    private function getParameter($param)
+    {
+        return $this->params->get($param);
+    }
+
+    private function isSvgLogo($nid): bool
+    {
+        $branding = $this->getParameter('branding');
+        return in_array($nid, $branding['logos']['filetypes']['svg']);
```
Comment (Tasiobg) [inline src/Twig/NetworkLogoExtension.php:67]:
```suggestion
        return \in_array($nid, $branding['logos']['filetypes']['svg']);
```
enable some opcode optimization, details and bechmark if you feel curiosity https://github.com/Roave/FunctionFQNReplacer

File: src/Twig/NetworkLogoExtension.php [hunk 4]
```diff
@@ -69,10 +44,33 @@ public function getNetworkLogo(string $nid, string $size): string
...
+    }
+
+    private function isAllowedLogo($nid): bool
+    {
+        $branding = $this->getParameter('branding');
+        $filetypes = $branding['logos']['filetypes'];
+        return in_array($nid, $filetypes['png']) || in_array($nid, $filetypes['svg']);
```
Comment (Tasiobg) [inline src/Twig/NetworkLogoExtension.php:74]:
```suggestion
        return \in_array($nid, $filetypes['png']) || in_array($nid, $filetypes['svg']);
```

Comment (gmarinov) [reply src/Twig/NetworkLogoExtension.php:74]: that would need two FQNs, one for each `in_array` .. also TIL.

File: tests/Branding/LogoAssetTest.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+namespace Tests\App\Branding;
+
+// this test works with the local asset files in /assets (not S3 or web/assets)
+
+use Tests\App\BaseWebTestCase;
+
+class LogoAssetTest extends BaseWebTestCase
+{
+    protected $logos = null;
+    const ASSET_PATH = 'assets/images/logos';
```
Comment (Tasiobg) [inline tests/Branding/LogoAssetTest.php:12]:
```suggestion
    private const ASSET_PATH = 'assets/images/logos';
```
Code sniffer says that visibility must be declared after PHP 7.1
I don't think this matter in a test but not hurting to have it

**Pull request merged**